src/index.rs

```diff
@@ -341,11 +369,16 @@ impl ResourceIndex {
         Ok(IndexUpdate { deleted, added })
     }
 
-    // the caller must ensure that:
+    // The caller must ensure that:
     // * the index is up-to-date except this single path
     // * the path hasn't been indexed before
```

Comment:
Let's make these [doc comments](https://doc.rust-lang.org/reference/comments.html#doc-comments)
just change "//" to "///" before all functions